pkg/controller/storagecluster/reconcile.go

```diff
@@ -425,15 +425,13 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 				return fmt.Errorf("failed to validate StorageDeviceSet %d: no StorageClass specified for walPVCTemplate", i)
 			}
 		}
+
 		if ds.DeviceType != "" {
-			if (DeviceTypeSSD == strings.ToLower(ds.DeviceType)) || (DeviceTypeHDD == strings.ToLower(ds.DeviceType)) || (DeviceTypeNVMe == strings.ToLower(ds.DeviceType)) {
-				metav1.SetMetaDataAnnotation(&sc.ObjectMeta, "crushDeviceClass", ds.DeviceType)
-			} else {
+			if (DeviceTypeSSD != strings.ToLower(ds.DeviceType)) || (DeviceTypeHDD != strings.ToLower(ds.DeviceType)) || (DeviceTypeNVMe != strings.ToLower(ds.DeviceType)) {
```

Comment:
The conditions should be && instead of ||

Reply:
I was confused about it but after checking through all conditions && is the appropriate operator . Adding || will fail the condition. Done